app/src/main/java/com/example/qrchive/Fragments/ScanResultPopupFragment.java

```diff
@@ -76,15 +83,14 @@ public void onClick(DialogInterface dialog, int which) {
         return builder.create();
     }
 
-    public ScanResultPopupFragment(FirebaseWrapper fbw) {
+    public ScanResultPopupFragment(FirebaseWrapper fbw, Fragment scanFrag) {
         this.fbw = fbw;
+        this.scanFrag = scanFrag;
     }
 
 
     private void sendResult(){
 
-        Fragment scanFrag = getParentFragment();
```

Comment:
This was a HUGE problem. The parent fragment WAS NOT scanFragment  and was ALWAYS null, hence the code would just return. So instead of getting the scanFrag like this, I made the class store a reference to scanFrag.